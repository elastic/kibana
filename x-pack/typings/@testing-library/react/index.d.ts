/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

// Write custom types because @types/testing-library__react doesn't work with our version of TypeScript
declare module '@testing-library/react' {
  // Lifted from https://github.com/testing-library/dom-testing-library/blob/4b2976d5161aa6ae3f35e5f9a9952e63c1997768/types/wait.d.ts
  export function wait(
    callback?: () => void,
    options?: {
      timeout?: number;
      interval?: number;
    }
  ): Promise<void>;
}
