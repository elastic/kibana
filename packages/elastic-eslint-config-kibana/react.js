const semver = require('semver')
const PKG = require('../../package.json')

module.exports = {
  plugins: [
    'react',
    'react-hooks',
    'jsx-a11y',
  ],

  parserOptions: {
    ecmaFeatures: {
      jsx: true
    }
  },

  settings: {
    react: {
      version: semver.valid(semver.coerce(PKG.dependencies.react)),
    },
  },

  rules: {
    'react/jsx-uses-react': 'error',
    'react/react-in-jsx-scope': 'error',
    'react/jsx-uses-vars': 'error',
    'react/jsx-no-undef': 'error',
    'react/jsx-pascal-case': 'error',
    'react/jsx-no-duplicate-props': ['error', { ignoreCase: true }],
    'react/no-danger': 'error',
    'react/self-closing-comp': 'error',
    'react-hooks/rules-of-hooks': 'error', // Checks rules of Hooks
    'react-hooks/exhaustive-deps': 'error', // Checks effect dependencies
    'jsx-a11y/accessible-emoji': 'error',
    'jsx-a11y/alt-text': 'error',
    'jsx-a11y/anchor-has-content': 'error',
    'jsx-a11y/aria-activedescendant-has-tabindex': 'error',
    'jsx-a11y/aria-props': 'error',
    'jsx-a11y/aria-proptypes': 'error',
    'jsx-a11y/aria-role': 'error',
    'jsx-a11y/aria-unsupported-elements': 'error',
    'jsx-a11y/click-events-have-key-events': 'error',
    'jsx-a11y/heading-has-content': 'error',
    'jsx-a11y/html-has-lang': 'error',
    'jsx-a11y/iframe-has-title': 'error',
    'jsx-a11y/interactive-supports-focus': 'error',
    'jsx-a11y/label-has-associated-control': 'error',
    'jsx-a11y/media-has-caption': 'error',
    'jsx-a11y/mouse-events-have-key-events': 'error',
    'jsx-a11y/no-access-key': 'error',
    'jsx-a11y/no-distracting-elements': 'error',
    'jsx-a11y/no-interactive-element-to-noninteractive-role': 'error',
    'jsx-a11y/no-noninteractive-element-interactions': 'error',
    'jsx-a11y/no-noninteractive-element-to-interactive-role': 'error',
    'jsx-a11y/no-onchange': 'error',
    'jsx-a11y/no-redundant-roles': 'error',
    'jsx-a11y/role-has-required-aria-props': 'error',
    'jsx-a11y/role-supports-aria-props': 'error',
    'jsx-a11y/scope': 'error',
    'jsx-a11y/tabindex-no-positive': 'error',
    'react/no-will-update-set-state': 'error',
    'react/no-is-mounted': 'error',
    'react/no-multi-comp': ['error', { ignoreStateless: true }],
    'react/no-unknown-property': 'error',
    'react/prefer-es6-class': ['error', 'always'],
    'react/prefer-stateless-function': ['error', { ignorePureComponents: true }],
    'react/no-unescaped-entities': 'error',
  },
}
