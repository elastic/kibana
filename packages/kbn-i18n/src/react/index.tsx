/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { InjectedIntl as _InjectedIntl, InjectedIntlProps as _InjectedIntlProps } from 'react-intl';

export type InjectedIntl = _InjectedIntl;
export type InjectedIntlProps = _InjectedIntlProps;

export {
  intlShape,
  FormattedDate,
  FormattedTime,
  FormattedRelative,
  FormattedNumber,
  FormattedPlural,
  FormattedMessage,
  FormattedHTMLMessage,
  // Only used for testing. Use I18nProvider otherwise.
  IntlProvider as __IntlProvider,
} from 'react-intl';

export { I18nProvider } from './provider';
export { injectI18n } from './inject';
