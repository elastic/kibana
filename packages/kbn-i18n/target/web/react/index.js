"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "intlShape", {
  enumerable: true,
  get: function get() {
    return _reactIntl.intlShape;
  }
});
Object.defineProperty(exports, "InjectedIntl", {
  enumerable: true,
  get: function get() {
    return _reactIntl.InjectedIntl;
  }
});
Object.defineProperty(exports, "FormattedDate", {
  enumerable: true,
  get: function get() {
    return _reactIntl.FormattedDate;
  }
});
Object.defineProperty(exports, "FormattedTime", {
  enumerable: true,
  get: function get() {
    return _reactIntl.FormattedTime;
  }
});
Object.defineProperty(exports, "FormattedRelative", {
  enumerable: true,
  get: function get() {
    return _reactIntl.FormattedRelative;
  }
});
Object.defineProperty(exports, "FormattedNumber", {
  enumerable: true,
  get: function get() {
    return _reactIntl.FormattedNumber;
  }
});
Object.defineProperty(exports, "FormattedPlural", {
  enumerable: true,
  get: function get() {
    return _reactIntl.FormattedPlural;
  }
});
Object.defineProperty(exports, "FormattedMessage", {
  enumerable: true,
  get: function get() {
    return _reactIntl.FormattedMessage;
  }
});
Object.defineProperty(exports, "FormattedHTMLMessage", {
  enumerable: true,
  get: function get() {
    return _reactIntl.FormattedHTMLMessage;
  }
});
Object.defineProperty(exports, "I18nProvider", {
  enumerable: true,
  get: function get() {
    return _provider.I18nProvider;
  }
});
Object.defineProperty(exports, "injectI18n", {
  enumerable: true,
  get: function get() {
    return _inject.injectI18n;
  }
});

var _reactIntl = require("react-intl");

var _provider = require("./provider");

var _inject = require("./inject");