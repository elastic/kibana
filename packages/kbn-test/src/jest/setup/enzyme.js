/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * and the Server Side Public License, v 1; you may not use this file except in
 * compliance with, at your election, the Elastic License or the Server Side
 * Public License, v 1.
 */

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
