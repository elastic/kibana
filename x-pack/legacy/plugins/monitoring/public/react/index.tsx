/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import { I18nContext } from 'ui/i18n';
import { Route, Switch, HashRouter } from 'react-router-dom';

import { NoDataView } from './views/';

export const App: React.FC = () => {
  return (
    <I18nContext>
      <HashRouter>
        <Switch>
          <Route exact path={`/no-data`} component={NoDataView} />
        </Switch>
      </HashRouter>
    </I18nContext>
  );
};
