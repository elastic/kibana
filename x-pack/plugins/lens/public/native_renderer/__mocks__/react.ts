/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';

// This mock overwrites react to use the layout effect hook instead of the normal one.
// This is done to have effects executed synchronously to be able to test them without
// setTimeout hacks.

// eslint-disable-next-line import/no-default-export
export default { ...React, useEffect: React.useLayoutEffect };
