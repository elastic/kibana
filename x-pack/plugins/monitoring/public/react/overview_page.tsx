/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { EuiErrorBoundary } from '@elastic/eui';
import React, { useState, useEffect, useCallback } from 'react';
import moment from 'moment';
import { useKibana } from '../../../../../src/plugins/kibana_react/public';
import { Overview } from '../components/cluster/overview';

import { CODE_PATH_ALL } from '../../common/constants';

const CODE_PATHS = [CODE_PATH_ALL];

export const OverviewPage = () => {
  const [data, setData] = useState(undefined);
  const kibana = useKibana();

  useEffect(() => {
    // from services/clusters.js
    const fetchClusters = async () => {
      const url = '/api/monitoring/v1/clusters/dltQ_pDUTOW04Do08ltpdA';
      const max = moment();
      const min = moment().subtract(15, 'm');

      try {
        const response = await kibana.services.http.fetch(url, {
          method: 'POST',
          body: JSON.stringify({
            css: undefined,
            timeRange: {
              min: min.toISOString(),
              max: max.toISOString(),
            },
            codePaths: CODE_PATHS,
          }),
        });

        setData(response[0]);
      } catch (err) {
        // TODO: handle error
      }
    };

    fetchClusters();
  }, [kibana.services.http]);

  return (
    <EuiErrorBoundary>
      {data && (
        <Overview cluster={data} alerts={{}} setupMode={false} showLicenseExpiration={false} />
      )}
    </EuiErrorBoundary>
  );
};
