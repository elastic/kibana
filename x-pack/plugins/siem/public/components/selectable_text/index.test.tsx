/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import * as React from 'react';

import { SelectableText } from '.';

describe('SelectableText', () => {
  test('renders correctly against snapshot', () => {
    const wrapper = shallow(<SelectableText>You may select this text</SelectableText>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('it applies the user-select: text style', () => {
    const wrapper = mount(<SelectableText>You may select this text</SelectableText>);

    expect(wrapper).toHaveStyleRule('user-select', 'text');
  });
});
