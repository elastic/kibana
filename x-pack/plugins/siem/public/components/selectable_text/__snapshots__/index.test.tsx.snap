// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectableText renders correctly against snapshot 1`] = `
.c0 {
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
}

<span
  className="c0"
>
  You may select this text
</span>
`;
